README.md

```diff
@@ -0,0 +1,111 @@
+# stream-json-rpc
+Call remote process methods easily, using the transport that you need.
+
+## Usage
+Example using node-ipc
+```typescript
+// The transport layer must implement the `stream.Duplex` interface
+// Here are the ones for `node-ipc`
+import { Duplex } from 'stream';
+
+class ServerDuplex extends Duplex {
+  ipc: NodeIPC.Server; // instance of nodeipc.server
+  socket: NodeIPC.Socket; // destination socket
+
+  constructor(ipc: NodeIPC.Server, socket: NodeIPC.Socket) {
+    super();
+    this.ipc = ipc;
+    this.socket = socket;
+
+    ipc.on('data', data => {
+      this.push(data);
+    });
+  }
+
+  // tslint:disable-next-line
+  _write(chunk: any, _encoding: any, callback: any) {
+    this.ipc.emit(this.socket, chunk);
+    callback();
+  }
+
+  // tslint:disable-next-line
+  _read(_size: any) {}
+}
+
+class ClientDuplex extends Duplex {
+  ipc: NodeIPC.Client; // ipc socket opened between current process and the server
+
+  constructor(ipc: NodeIPC.Client) {
+    super();
+    this.ipc = ipc;
+
+    ipc.on('data', data => {
+      this.push(data);
+    });
+  }
+
+  // tslint:disable-next-line
+  _write(chunk: any, _encoding: any, callback: any) {
+    this.ipc.emit(chunk);
+    callback();
+  }
+
+  // tslint:disable-next-line
+  _read(_size: any) {}
+}
+```
```

Comment:
This would have its place in a dedicated library to transform node-ipc to a stream

Reply:
👍 and some for the `electron.ipc` (probably with multiplexing).

Reply:
This repo could hold several module and be published via lerna